core/vtk/ttkBarycentricSubdivision/ttkBarycentricSubdivision.cpp

```diff
@@ -1,21 +1,36 @@
 #include <ttkBarycentricSubdivision.h>
+#include <ttkMacros.h>
 #include <ttkUtils.h>
+
 #include <vtkCellData.h>
+#include <vtkInformation.h>
 #include <vtkPointData.h>
+#include <vtkUnstructuredGrid.h>
+
+vtkStandardNewMacro(ttkBarycentricSubdivision);
 
-#define MODULE_S "[ttkBarycentricSubdivision] "
-#define MODULE_ERROR_S MODULE_S "Error: "
-#ifndef TTK_ENABLE_KAMIKAZE
-#define TTK_ABORT_KK(COND, MSG, RET)    \
-  if(COND) {                            \
-    cerr << MODULE_ERROR_S MSG << endl; \
-    return RET;                         \
+ttkBarycentricSubdivision::ttkBarycentricSubdivision() {
+  SetNumberOfInputPorts(1);
+  SetNumberOfOutputPorts(1);
+}
+
+int ttkBarycentricSubdivision::FillInputPortInformation(int port,
+                                                        vtkInformation *info) {
+  if(port == 0) {
+    info->Set(vtkAlgorithm::INPUT_REQUIRED_DATA_TYPE(), "vtkUnstructuredGrid");
```

Comment:
actually, I'd suggest to go for ```vtkDataSet``` instead of ```vtkUnstructuredGrid```, to also support implicitly triangulated ```vtkImageData```, which can often be useful.

Reply:
Addressed in 30ed8f3